### ISSUE TITLE
No reconciliation after changing podTargetLabels

### ISSUE THREAD

**Issue by gkocur** [infinispan/infinispan-operator#1390]
My use case: I want to add labels to pods to inject istio sidecar containers.
According to the documentation I added to the Infinispan manifest:
```
apiVersion: infinispan.org/v1
kind: Infinispan
metadata:
  name: infinispan
  annotations:
    infinispan.org/podTargetLabels: "sidecar.istio.io/inject"
  labels:
    sidecar.istio.io/inject: "true"
```

But nothing happened. I had to remove the statefulset completely and the new one was created with proper `.spec.template.metadata.labels`.

I'd rather expect the change in statefulset and the rolling update.

**Comment by ryanemerson** [comment]
@gkocur Thanks for raising the issue. We should be able to fix the reconciliation behaviour easily enough.

FYI, until https://github.com/infinispan/infinispan-operator/issues/1384 is merged, the operator won't work with istio unfortunately.

**Comment by gkocur** [comment]
Oh, didn't spot this issue about the istio, thank you.
I was able to inject the istio sidecar (by removing the statefulset), but it is still not tested (we'll need infinispan in the close future).

**Comment by tolleiv** [comment]
Just to confirm - this issue is still relevant with the most recent version 2.3.3.

**Comment by ryanemerson** [comment]
PR raised, this will be fixed in 2.3.7
